packages/fxa-auth-server/scripts/prune-tokens.ts

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import program from 'commander';
+import { BaseAuthModel } from 'fxa-shared/db/models/auth';
+import { StatsD } from 'hot-shots';
+import { knex } from 'knex';
+import moment from 'moment';
+const { PruneTokens } = require('fxa-shared/db/models/auth');
+const pckg = require('../package.json');
+
+/** Wrapper around moment to parse input args coming in as duration strings */
+function parseDuration(duration: string) {
+  return moment.duration(...duration.split(/\s|-/)).asMilliseconds();
+}
+
+export async function init() {
+  // Setup utilities
+  const config = require('../config').getProperties();
+  const statsd = new StatsD(config.statsd);
+  const log = require('../lib/log')(config.log.level, 'prune-tokens', statsd);
+
+  // Parse args
+  program
+    .version(pckg.version)
+    .option(
+      '--maxTokenAge <duration>',
+      'Max age of token. Any tokens older than this value will be pruned.',
+      '10 days'
+    )
+    .option(
+      '--maxCodeAge <duration>',
+      'Max age of code. Any codes older than this value will be pruned.',
+      '1 month'
+    )
+    .on('--help', function () {
+      console.log(`
+Example:
+
+> ./scripts/prune-tokens.sh --maxTokenAge='10 days' --maxCodeAge='1 month'
+
+Exit Codes:
+  0 - success
+  1 - unexpected error
+  2 - error during initialization
+  3 - error during pruning operation`);
+    })
+    .parse(process.argv);
+
+  const tokenMaxAge = parseDuration(program.maxTokenAge);
+  const codeMaxAge = parseDuration(program.maxCodeAge);
+
+  log.info('token pruning args', {
+    maxTokenAge: program.maxTokenAge,
+    maxCodeAge: program.maxCodeAge,
+  });
+
+  // Setup Knex Connection. PruneTokens relies on this being done.
+  try {
+    const instance = knex({
+      connection: {
+        typeCast: function (field: any, next: any) {
+          if (field.type === 'TINY' && field.length === 1) {
+            return field.string() === '1';
+          }
+          return next();
+        },
```

Comment:
nit: would be nice if we use the function defined in shared/db/index.ts here (and elsewhere where this function is copied).